File: Labs/Lab4/Solution.py
from TreeNode import TreeNode


class Solution:
    def __init__(self):
        print("Lab 4")

    def invertTree(self, root):
        if root is None:
            return root

        self.invertTree(root.left)
        self.invertTree(root.right)

        temp = root.left
        root.left = root.right
        root.right = temp

        return root

File: Labs/Lab4/Solution.java
public class Solution {
    public Solution () {}
    public TreeNode invertTree (TreeNode root) {
        if (root == null) return null;

        invertTree(root.left);
        invertTree(root.right);

        //swap
        TreeNode temp = root.right;
        root.right = root.left;
        root.left = temp;

        return root;
    }
}
